hooks/useGlobalHistoricalDebtData.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+import { orderBy } from 'lodash';
+import useDailyIssuedQuery from 'queries/staking/useDailyIssuedQuery';
+import useDailyBurnedQuery from 'queries/staking/useDailyBurnedQuery';
+import { StakingTransactionType } from 'queries/staking/types';
+
+type HistoricalGlobalDebtAndIssuanceData = {
+	issuance: number;
+	debtPool: number;
+	timestamp: number;
+};
+
+type HistoricalGlobalDebtAndIssuance = {
+	isLoading: boolean;
+	data: HistoricalGlobalDebtAndIssuanceData[];
+};
+
+const useGlobalHistoricalDebtData = () => {
+	const [historicalDebt, setHistoricalDebt] = useState<HistoricalGlobalDebtAndIssuance>({
+		isLoading: true,
+		data: [],
+	});
+
+	const issuedQuery = useDailyIssuedQuery();
+	const burnedQuery = useDailyBurnedQuery();
+
+	const isLoaded = issuedQuery.isSuccess && burnedQuery.isSuccess;
+
+	useEffect(() => {
+		if (isLoaded) {
+			const issued = issuedQuery.data ?? [];
+			const burned = burnedQuery.data ?? [];
+
+			// We concat both the events and order them (asc)
+			const eventBlocks = orderBy(burned.concat(issued), 'timestamp', 'asc');
+
+			const data: HistoricalGlobalDebtAndIssuanceData[] = [];
+			let debtPoolStartingValue: number = 0;
+
+			eventBlocks.forEach((event, i) => {
+				const multiplier = event.type === StakingTransactionType.Burned ? -1 : 1;
+				const aggregation =
+					data.length === 0
+						? multiplier * event.value
+						: multiplier * event.value + data[i - 1].issuance;
+				if (data.length === 0) {
+					debtPoolStartingValue = event.totalDebt;
+				}
+
```

Comment:
Logic LGTM but I've got a couple of comments. I'm curious to see what you think:

- So because you'll get the debt pool value from let's say a year ago, and will start your aggregation from 0 (or at least with the first burn/mint event value) I was thinking we could "hack" the data and set `issuance` to be equal to `debtPoolStartingValue`. Then both lines will start from the same point and we will see diverge at some point in `t + n`. Thought?

- Also I'm not sure what's the logic behind `debtPool: event.totalDebt - debtPoolStartingValue`. Are you subtracting  `debtPoolStartingValue` from `event.totalDebt` every time?

Reply:
- The first point makes sense, I can try that!
- I wanted to normalize all of the `totalDebt` values to whatever the initial value was at the start of the time range, so the graph would actually show the change in total debt starting from the earliest data point. The goal was to put `totalDebt` values on the same scale as issued/burned so the comparison made more sense, but I'm not sure if this actually accomplishes that